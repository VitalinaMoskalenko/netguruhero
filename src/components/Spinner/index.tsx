import React from "react";
import styled from "styled-components";

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex: 1;
`;

const StyledSpinner = styled.svg`
  animation: rotate 2s linear infinite;
  width: 50px;
  height: 50px;

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }
  @keyframes dash {
    0% {
      stroke-dasharray: 1, 150;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -124;
    }
  }
`;

const Circle = styled.circle`
  stroke: ${({ theme }) => theme.colors.green};
  stroke-linecap: round;
  animation: dash 1.5s ease-in-out infinite;
`;

const Spinner = () => (
  <Container>
    <StyledSpinner viewBox="0 0 50 50">
      <Circle cx="25" cy="25" r="20" fill="none" strokeWidth="4" />
    </StyledSpinner>
  </Container>
);

export default Spinner;
