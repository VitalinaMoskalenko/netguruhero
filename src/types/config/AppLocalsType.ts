export enum AppLocalsType {
  en = "en",
}
