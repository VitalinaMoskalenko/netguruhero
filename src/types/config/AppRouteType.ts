export enum AppRouteType {
  home = "/",
  addHeroForm = "/addHeroForm",
  heroDetails = "/heroDetails",
  heroDetailsWithId = "/heroDetails/:id",
}
