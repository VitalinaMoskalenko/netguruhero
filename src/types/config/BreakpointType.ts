export enum BreakpointType {
  xxs = 375,
  xs = 768,
  xxl = 1440,
}
