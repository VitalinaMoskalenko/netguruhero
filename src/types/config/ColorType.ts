export enum ColorType {
  white = "white",
  black = "black",
  green = "#63cd93",
  blue = "#6199f3",
  gray = "#f5f6fb",
  black05 = "rgba(0,0,0, 0.5)",
}
