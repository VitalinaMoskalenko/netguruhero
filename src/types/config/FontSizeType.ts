export enum FontSizeType {
  smallBody = 14,
  body = 18,
  h5 = 16,
  h4 = 18,
  h3 = 24,
  h2 = 32,
  h1 = 48,
}
