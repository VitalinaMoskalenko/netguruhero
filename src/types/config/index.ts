export * from "./AppRouteType";
export * from "./FontSizeType";
export * from "./ColorType";
export * from "./BreakpointType";
export * from "./AppLocalsType";
