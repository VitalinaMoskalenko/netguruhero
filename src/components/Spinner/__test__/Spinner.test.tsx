import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import Spinner from "../";
import { theme } from "../../../config";

it("renders spinner correctly", () => {
  const spinnerComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Spinner />
      </ThemeProvider>
    )
    .toJSON();

  expect(spinnerComponent).toMatchSnapshot();
});
