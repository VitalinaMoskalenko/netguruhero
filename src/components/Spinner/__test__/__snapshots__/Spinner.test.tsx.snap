// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders spinner correctly 1`] = `
<div
  className="sc-bdnxRM jgTusu"
>
  <svg
    className="sc-gtsrHT fXTByD"
    viewBox="0 0 50 50"
  >
    <circle
      className="sc-dlnjwi bSwSsA"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      strokeWidth="4"
    />
  </svg>
</div>
`;
